### PR TITLE
Feat/delete branch on slide out

### PR DESCRIPTION
**Pull request by mkondratek**
This minor improvement introduces a ~registry~ git config key which then decides whether slid out local branch should be deleted. 
key name: `machete.slideout.deletelocalbranch`
turn it on: `git config --global --add machete.slideout.deletelocalbranch true`

### CODE REVIEW

**Comment by mkondratek** [comment]
Whoops, it turned out that the specific order execution of tasks is hard to achieve.
In this case, scheduling graph table refresh after branch deletion. 
The solution is available since 2020.1.

**Comment by PawelLipski** [comment]
@mkondratek leaving the assignment of this task to a milestone up to you, depending on how you see the amount of work that's left to complete

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java [hunk 1]
```diff
@@ -20,6 +29,8 @@
       IExpectsKeyGitMacheteRepository,
       IExpectsKeyProject {
 
+  private static final String DELETE_LOCAL_BRANCH_ON_SLIDE_OUT_GIT_CONFIG_KEY = "machete.slideout.deletelocalbranch";
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:32]: Nit: git config keys are case-INsensitive, but for clarity:
```suggestion
  private static final String DELETE_LOCAL_BRANCH_ON_SLIDE_OUT_GIT_CONFIG_KEY = "machete.slideOut.deleteLocalBranch";
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java [hunk 2]
```diff
@@ -99,5 +108,56 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
...
+        var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+            .map(b -> b.equals(branchName))
+            .getOrElse(true);
+        if (slidOutBranchIsCurrent) {
+          LOG.warn("Skipping local branch deletion because it is equal to current branch");
+        }
+
+        var branchesToContainingRepositories = java.util.Collections.<String, java.util.List<? extends GitRepository>>singletonMap(
+            branchName,
+            selectedVcsRepository.toJavaList());
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:134]: Join these lines (if autoformatter allow that)

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java [hunk 3]
```diff
@@ -99,5 +108,56 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
...
+          DELETE_LOCAL_BRANCH_ON_SLIDE_OUT_GIT_CONFIG_KEY);
+      var value = ApplicationManager.getApplication().runReadAction(computable);
+      boolean result = false;
+      if (value != null) {
+        Boolean booleanValue = GitConfigUtil.getBooleanValue(value);
+        result = booleanValue != null && booleanValue;
+      }
+      return result;
+    } catch (VcsException e) {
+      // ignore
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:158]: Maybe write something to log?

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java (hunk 2 above)
Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:134]: it's autoformatter :confused:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java (hunk 3 above)
Comment (mkondratek) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java]: done

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java [hunk 4]
```diff
@@ -99,5 +108,57 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
...
+      var shallDeleteLocalBranch = getDeleteLocalBranchOnSlideOutGitConfigKeyValue(root, project);
+      if (shallDeleteLocalBranch) {
+        var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+            .map(b -> b.equals(branchName))
+            .getOrElse(true);
+        if (slidOutBranchIsCurrent) {
+          LOG.warn("Skipping local branch deletion because it is equal to current branch");
+        }
+
+        var branchesToContainingRepositories = java.util.Collections.<String, java.util.List<? extends GitRepository>>singletonMap(
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:132]: Maybe instead sth like:

```suggestion
        Map<String, java.util.List<? extends GitRepository>> branchesToContainingRepositories = java.util.Collections.singletonMap(
```

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java [hunk 5]
```diff
@@ -99,5 +108,57 @@ private void doSlideOut(AnActionEvent anActionEvent, IGitMacheteNonRootBranch br
...
+    if (selectedVcsRepository.isDefined()) {
+      var root = selectedVcsRepository.get().getRoot();
+      var project = getProject(anActionEvent);
+      var shallDeleteLocalBranch = getDeleteLocalBranchOnSlideOutGitConfigKeyValue(root, project);
+      if (shallDeleteLocalBranch) {
+        var slidOutBranchIsCurrent = getCurrentBranchNameIfManaged(anActionEvent)
+            .map(b -> b.equals(branchName))
+            .getOrElse(true);
+        if (slidOutBranchIsCurrent) {
+          LOG.warn("Skipping local branch deletion because it is equal to current branch");
```
Comment (PawelLipski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutBranchAction.java:129]: :+1:

**Pull request merged**